src/mito2/src/worker/handle_drop.rs

```diff
@@ -0,0 +1,136 @@
+            .write(&marker_path, vec![])
+            .await
+            .context(OpenDalSnafu)?;
+
+        // remove this region from region map to prevent other requests from accessing this region
+        self.regions.remove_region(region_id);
+        self.dropping_regions.insert_region(region.clone());
+
+        // mark region version as dropped
+        region.version_control.mark_dropped();
+        info!(
+            "Region {} is dropped logically, but some files are not deleted yet",
+            region_id
+        );
+
+        // detach a background task to delete the region dir
+        let region_dir = region.region_dir.clone();
+        let object_store = self.object_store.clone();
+        let dropping_regions = self.dropping_regions.clone();
+        common_runtime::spawn_bg(async move {
+            later_drop_task(region_id, region_dir, object_store, dropping_regions).await;
+        });
+
+        Ok(Output::AffectedRows(0))
+    }
+}
+
+/// Background GC task to remove the entire region path once it find there is no
+/// parquet file left.
+///
+/// This task will keep running until finished. Any resource captured by it will
+/// not be released before then. Be sure to only pass weak reference if something
+/// is depended on ref-count mechanism.
+async fn later_drop_task(
+    region_id: RegionId,
+    region_path: String,
+    object_store: ObjectStore,
+    dropping_regions: RegionMapRef,
+) {
+    loop {
+        let result: Result<()> = try {
+            sleep(Duration::from_secs(GC_TASK_INTERVAL_SEC)).await;
+
+            // list all files under the given region path to check if there are un-deleted parquet files
+            let mut has_parquet_file = false;
+            // record all paths that neither ends with .parquet nor the marker file
+            let mut files_to_remove_first = vec![];
+            let mut files = object_store
+                .list(&region_path)
+                .await
+                .context(OpenDalSnafu)?;
+            while let Some(file) = files.next().await {
+                let file = file.context(OpenDalSnafu)?;
+                if file.path().ends_with(".parquet") {
+                    has_parquet_file = true;
+                    break;
+                } else if !file.path().ends_with(DROPPING_MARKER_FILE) {
+                    files_to_remove_first.push(file.path().to_string());
+                }
+            }
+
+            if !has_parquet_file {
+                // no parquet file found, delete the region path
+                // first delete all files other than the marker
+                try_join_all(
+                    files_to_remove_first
+                        .iter()
+                        .map(|path| object_store.delete(path)),
+                )
+                .await
+                .context(OpenDalSnafu)?;
+                // then remove the marker with this dir
+                object_store
+                    .delete(&region_path)
```

Comment:
Will `delete()` remove the whole directory recursively? There is a [remove_all()](https://docs.rs/opendal/latest/opendal/struct.Operator.html#method.remove_all) method.